### PR TITLE
Kurtwheeler/alternate accession codes

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

https://github.com/AlexsLemonade/refinebio/issues/559

## Purpose/Implementation Notes

Makes it so that searching on `GSM1234` will match experiments with accession code `GSM1234` OR `E-GEOD-1234`. This won't help if someone searches `/experiment/?accession_code=GSM1234` though.

It does so by adding an `alternate_accession_code` to the experiment model which is included in searches and is set via the Experiment.save() method.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

Unit tests.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: common/data_refinery_common/models/models.py [hunk 1]
```diff
@@ -279,6 +280,13 @@ def save(self, *args, **kwargs):
         if not self.id:
             self.created_at = current_time
         self.last_modified = current_time
+
+        if self.accession_code and not self.alternate_accession_code:
+            if self.accession_code.startswith('GSM'):
+                self.alternate_accession_code = 'E-GEOD-' + self.accession_code[3:]
+            elif self.accession_code.startswith('E-GEOD-'):
+                self.alternate_accession_code = 'GSM' + self.accession_code[7:]
```
Comment (Miserlou) [inline common/data_refinery_common/models/models.py]: This file uses `GSM` and the migration uses `GSE`. What's going on with that? Should that be tested?

**Comment by kurtwheeler** [comment]
In an automated way? Looks like we could use https://github.com/plumdog/django_migration_testcase which was inspired by https://www.caktusgroup.com/blog/2016/02/02/writing-unit-tests-django-migrations/. The project hasn't been touched in over a year. I can try it though.

**Comment by kurtwheeler** [comment]
I do think that a test on the staging server will be sufficient. I'll see if I can get django-migration-testcase working before we wanna do a staging deploy, and if not we'll just let that test it.

**Comment by kurtwheeler** [comment]
Ok, I tried to write a migration test using that library. I got this exception:
```
ERROR: test_migration (data_refinery_api.tests.AltAccessionMigrationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django_migration_testcase/django_migrations.py", line 16, in setUp
    self.migrate(app_name, version)
  File "/usr/local/lib/python3.6/site-packages/django_migration_testcase/base.py", line 89, in migrate
    call_command(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 140, in call_command
    ', '.join(sorted(valid_options)),
TypeError: Unknown option(s) for migrate command: no_initial_data. Valid options are: app_label, database, fake, fake_initial, help, interactive, migration_name, no_color, no_input, pythonpath, run_syncdb, settings, skip_checks, stderr, stdout, traceback, verbosity, version.
```
which seems to be caused by this option being passed through: https://github.com/plumdog/django_migration_testcase/blob/master/django_migration_testcase/base.py#L78

However I think that's as far as it's worth it to debug since we can test this on staging and this package doesn't seem to be maintained since I'm the first one running into that.

**Comment by Miserlou** [comment]
Ok!

**Pull request merged**